compiler/noirc_evaluator/src/ssa/opt/databus.rs

```diff
@@ -0,0 +1,46 @@
+use crate::ssa::ir::instruction::Instruction;
+
+
+
+impl Ssa {
+    /// Map arrays with the last instruction that uses it
+    /// For this we simply process all the instructions in execution order
+    /// and update the map whenever there is a match
+    pub(crate) fn map_to_call_data_array(&self) -> HashMap<ValueId, InstructionId> {
+        let mut array_use = HashMap::default();
+        for func in self.functions.values() {
+            let mut reverse_post_order = PostOrder::with_function(func).into_vec();
+            reverse_post_order.reverse();
+            for block in reverse_post_order {
+                last_use(block, &func.dfg, &mut array_use);
+            }
+        }
+        array_use
+    }
+//la question c comment on remplace  ?
```

Comment:
Some leftover french here

Reply:
that's an old file that was left-over, I removed it.